### DISCUSSION TITLE
How to extract JUST GPS data ?

### DISCUSSION THREAD

**Discussion by JHenry83** [ImageMagick/ImageMagick#4806] [General]
I am using magick  identify -verbsose {imagename.jpg} > {imagenametextfile.,txt} to get the EXIF GPS LAT and LONG data from several images. I get the GPS data as well as a LOT of metadata that I do not need. Is there a switch or some command modifier that will give me just the GPS data?

**Comment by fmw42** [comment]
No flag for GPS only. But you can create a command for any EXIF or IPTC field or fields from string formats. See https://imagemagick.org/script/escape.php

Post and example image and the GPS fields you need and some one can try to create a command to help you.

**Reply by JHenry83** [reply]
> No flag for GPS only. But you can create a command for any EXIF or IPTC field or fields from string formats. See https://imagemagick.org/script/escape.php
> 
> Post and example image and the GPS fields you need and some one can try to create a command to help you.

Thank you fmv42 for you quick reply. Before I post an example image I am going to read over that link that you posted. 

[https://imagemagick.org/script/escape.php](url)

To see if I can figure out how to do it on my own, I always like to explore and try things on my own first and then if I can not come up with a solution, then ask for help. I may be back in a few days with another question. Thanks again.

**Reply by fmw42** [reply]
For ImageMagick 7,

`magick image.suffix -format "%[EXIF:field]" info:`

for Imagemgick 6, change magick to convert.

You an put multiple fields there also by repeating %[EXIF:field] in the quotes.

**Reply by JHenry83** [reply]
That command leaves out identify will imagemagick still look for the image metadata without that?

**Reply by fmw42** [reply]
Yes

**Reply by JHenry83** [reply]
Ok I am totally confused here is a sample file.
![Beauty](https://user-images.githubusercontent.com/98428774/152626751-8d3ea75f-677a-4eb3-8b10-9eca0c94c809.jpg)
I run --- magick identify -verbose beauty.jpg > beautyjpg.txt and get this:
[beautyjpg.txt](https://github.com/ImageMagick/ImageMagick/files/8007277/beautyjpg.txt)
 As you can see there is a line in there call exif:GPSLatitude:
but if I follow what you said and do 

magick beauty.jpg %[exif:Latitude:] info

I get this
magick: unable to open image '%[exif:Latitude:]': Invalid argument @ error/blob.c/OpenBlob/3527.
magick: no decode delegate for this image format `' @ error/constitute.c/ReadImage/572.

**Reply by fmw42** [reply]
It is GPSLatitude with no ending colon. Also you need -format

`magick beauty.jpg -format "%[exif:GPSLatitude]" info:`

32/1, 43/1, 222079/10000

**Reply by JHenry83** [reply]
fmw42 Thank you so much. that works, Now I have t cross my fingers and hope that the version of imagemagick that Firebase runs online for the Firebase Cloud Functions service includes this command or allows it. 

On a personal note thank you for your quick and respectful response. I subscribe to a LOT of technical IT forums and a few freenode.net IRC channels. Sometimes when a beginner ask a simple question like this the experts get arrogant and nasty. They seem to forget that they were once newbies themselves.

**Reply by Jmuccigr** [reply]
Any reason not to use exiftool instead, if all you're after is the GPS data?
```
> exiftool -s -*GPS* 152626751-8d3ea75f-677a-4eb3-8b10-9eca0c94c809.jpg
GPSVersionID                    : 2.2.0.0
GPSLatitudeRef                  : North
GPSLongitudeRef                 : West
GPSAltitudeRef                  : Above Sea Level
GPSTimeStamp                    : 01:19:34
GPSProcessingMethod             : GPS
GPSDateStamp                    : 2022:01:15
GPSAltitude                     : 29 m Above Sea Level
GPSDateTime                     : 2022:01:15 01:19:34Z
GPSLatitude                     : 32 deg 43' 22.21" N
GPSLongitude                    : 117 deg 10' 12.93" W
GPSPosition                     : 32 deg 43' 22.21" N, 117 deg 10' 12.93" W
```

**Reply by JHenry83** [reply]
I have EXIFTOOL on my local machine But I want to do this in the cloud. A user uploads a picture with my app which is hosted on Firebase then a cloud function runs which looks up the GPS coordinates to see if it is within a certain range. If so it is a valid picture If not reject it and tell me, the admin, and the user that it is not valid.. I have found a package on GitHub that supposedly looks at an image file's metadata that can be run as a cloud function. That package supposedly uses a version of ImageMagick that Firebase Storage runs natively. I am just looking to how to code it myself just in case that package does not do what I want it to do.

**Comment by snibgo** [comment]
You can use wildcards in IM's "-format", such as:
```
magick  traingps.jpg -format %[exif:gps*] info:

exif:GPSAltitude=29/1
exif:GPSAltitudeRef=0
exif:GPSDateStamp=2022:01:15
exif:GPSInfo=824
exif:GPSLatitude=32/1, 43/1, 222079/10000
exif:GPSLatitudeRef=N
exif:GPSLongitude=117/1, 10/1, 129272/10000
exif:GPSLongitudeRef=W
exif:GPSProcessingMethod=65, 83, 67, 73, 73, 0, 0, 0, 71, 80, 83, 0, 0, 0, 0, 0,
 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0
, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0
exif:GPSTimeStamp=1/1, 19/1, 34/1
exif:GPSVersionID=2, 2, 0, 0
```
